### PR TITLE
Added antismash params: pfam2go, rre, tfbs

### LINKED ISSUE #356: Add missing essential parameters for antismash
### Description of feature

The new version of antismash requires essential parametrs that i would suggest to be included in the pipeline. Those are `pfam2go --rre --tfbs` at the moment i add them using ext.args in the default runs.

### PR DESCRIPTION
**Pull request by jasmezz**
This closes #356. Changelog update should not be necessary for adding 3 params which could otherwise be given by `ext.args`.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/funcscan/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/funcscan _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -1176,13 +1176,34 @@
                     "fa_icon": "fas fa-search",
                     "enum": ["relaxed", "strict", "loose"]
                 },
+                "bgc_antismash_pfam2go": {
+                    "type": "boolean",
+                    "default": "false",
+                    "description": "Run Pfam to Gene Ontology mapping module.",
+                    "help_text": "For more information see the antiSMASH [documentation](https://docs.antismash.secondarymetabolites.org/).\n\n> Modifies tool parameter(s):\n> - antiSMASH: `--pfam2go`",
```
Comment (jfy133) [inline nextflow_schema.json:1183]: No further description what it does exactly? Why would you want to run ti?

File: nextflow_schema.json [hunk 2]
```diff
@@ -1176,13 +1176,34 @@
...
+                    "fa_icon": "fas fa-search"
+                },
+                "bgc_antismash_rre": {
+                    "type": "boolean",
+                    "default": "false",
+                    "description": "Run RREFinder precision mode on all RiPP gene clusters.",
+                    "help_text": "For more information see the antiSMASH [documentation](https://docs.antismash.secondarymetabolites.org/).\n\n> Modifies tool parameter(s):\n> - antiSMASH: `--rre`",
```
Comment (jfy133) [inline nextflow_schema.json:1190]: No further description what it does exactly? Why would you want to run it?

File: nextflow_schema.json [hunk 3]
```diff
@@ -1176,13 +1176,34 @@
...
                     "description": "Specify which taxonomic classification of input sequence to use.",
                     "help_text": "This specifies which set of secondary metabolites to screen for, based on the taxon type the secondary metabolites are from.\n\nThis will run different pipelines depending on whether the input sequences are from bacteria or fungi.\n\nFor more information see the antiSMASH [documentation](https://docs.antismash.secondarymetabolites.org/).\n\n> Modifies tool parameter(s):\n> - antiSMASH: `--taxon`",
                     "fa_icon": "fas fa-bacteria",
                     "enum": ["bacteria", "fungi"]
+                },
+                "bgc_antismash_tfbs": {
+                    "type": "boolean",
+                    "default": "false",
+                    "description": "Run TFBS finder on all gene clusters.",
+                    "help_text": "For more information see the antiSMASH [documentation](https://docs.antismash.secondarymetabolites.org/).\n\n> Modifies tool parameter(s):\n> - antiSMASH: `--tfbs`",
```
Comment (jfy133) [inline nextflow_schema.json:1205]: No further description what it does exactly? Why would you want to run it?

**Review by jfy133 — CHANGES_REQUESTED** [review]
Good to add the parameters, but they should actually be used somewhere, currently they don't do anything :grimacing: 

In addition:

- Changelog
- Do they produce extra output files? IF so should go in `output.md`

File: nextflow_schema.json (hunk 1 above)
Comment (jasmezz) [reply nextflow_schema.json:1183]: I have no clue why I want to run it :sweat_smile: There is no documentation other than this line in `antismash --help`. But I'll add the info that @Darcy220606 provided in the respective issue.

**Comment by jasmezz** [comment]
No new output files are produced, so no further docs necessary.

**Review by jfy133 — APPROVED** [review]

**Pull request merged**